app/src/main/java/dev/five_star/mybooks/booklist/MainViewModel.kt

```diff
@@ -54,9 +50,25 @@ class MainViewModel(private var repository: BookRepository) : ViewModel() {
         }
     }
 
+    private fun archiveBook(bookId: Int) {
+        viewModelScope.launch {
```

Comment:
I think Room will make sure that database operations happen off the main thread, but you could consider updating this to something like `viewModelScope.launch(Dispatches.IO){}` to be explicit about which thread the coroutine is running in